backend/controller/dal/notify.go

```diff
@@ -51,6 +53,88 @@ type event struct {
 	Deleted json.RawMessage `json:"deleted,omitempty"`
 }
 
+type deploymentState struct {
+	Key         model.DeploymentKey
+	schemaHash  []byte
+	minReplicas int
+}
+
+func deploymentStateFromDeployment(deployment Deployment) (deploymentState, error) {
+	hasher := sha256.New()
+	data := []byte(deployment.Schema.String())
+	if _, err := hasher.Write(data); err != nil {
+		return deploymentState{}, fmt.Errorf("failed to hash schema: %w", err)
+	}
```

Comment:
This is the hashing I had to change to. It seemed like `sha256.fromBytes` just didn't compute a different has at all, but maybe I was using it wrong.